nix/shell.sh

```diff
@@ -0,0 +1,105 @@
+echo nix/shell.sh: Entering a devShell
+export SRC_ROOT=$(pwd)
+export PATH=$SRC_ROOT/build/bin:$PATH
+
+banner()
+{
+    echo "+------------------------------------------+"
+    printf "| %-40s |\n" "$1"
+    echo "+------------------------------------------+"
+}
+
+
+function configure {
+    banner "Configuring with cmake"
+    cmake -S . -B./build \
+          -DBUILD_TESTING=ON \
+          -DS2N_INTEG_TESTS=ON \
+          -DS2N_INSTALL_S2NC_S2ND=ON \
+          -DS2N_FAST_INTEG_TESTS=ON \
+          -DBUILD_SHARED_LIBS=ON \
+          -DCMAKE_BUILD_TYPE=RelWithDebInfo
+}
+
+function build {
+    banner "Running Build"
+    javac tests/integrationv2/bin/SSLSocketClient.java
```

Comment:
This should really be added as a target in the cmake integ test section. I'm assuming that's possible?

Reply:
This simply reflects what the codebuild bin script does already. Moving the Java dependency into `CMakeLists.txt` is a goal but to do that idiomatically requires the use of ctests fixtures which requires at least cmake 3.7 (see [here](https://cmake.org/cmake/help/latest/prop_test/FIXTURES_REQUIRED.html)). Our minimum required cmake version is 3.0. One alternative to using fixtures is to create a script that cmake invokes (the script could handle the java compile, as well as apache setup, for the appropriate test cases).

I think this change is somewhat involved and should go in its own PR.